tests/unit/s2n_async_pkcs11_sign_test.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "testlib/s2n_testlib.h"
+
+#include <s2n.h>
+
+#include "error/s2n_errno.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_security_policies.h"
+#include "tls/s2n_cipher_suites.h"
+#include "utils/s2n_safety.h"
+
+#include <pthread.h>
+
+#define CK_PTR    *
+#define NULL_PTR    0
+#define CK_DEFINE_FUNCTION( returnType, name )             returnType name
+#define CK_DECLARE_FUNCTION( returnType, name )            returnType name
+#define CK_DECLARE_FUNCTION_POINTER( returnType, name )    returnType( CK_PTR name )
+#define CK_CALLBACK_FUNCTION( returnType, name )           returnType( CK_PTR name )
+
+/* Used to add SHA256 ASN1 encoding to the PKCS #11 RSA signature. */
+#define pkcs11STUFF_APPENDED_TO_RSA_SIG    { 0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04, 0x20 }
+
+/* Test config. */
+#define pkcs11_test_slot 2
+#define pkcs11_pin "0000"
+#define pkcs11_rsa_key_label "rsa-privkey"
+#define pkcs11_ec_key_label  "ecdsa-privkey"
+
+#include "pkcs11.h"
+
+struct s2n_async_pkey_op *pkey_op = NULL;
+static pthread_mutex_t pkcs11_mutex = {0};
+
+struct task_params {
+    struct s2n_connection *conn;
+    struct s2n_async_pkey_op *op;
+};
+
+static uint8_t verify_host_fn(const char *host_name, size_t host_name_len, void *data) {
+    return 1;
+}
```

Comment:
I doubt you care about certificate validation for this test. Consider using https://github.com/aws/s2n-tls/blob/main/tls/s2n_config.h#L128